compilers/concrete-compiler/compiler/lib/Runtime/wrappers.cpp

```diff
@@ -1011,3 +1012,20 @@ void memref_trace_message(char *message_ptr, uint32_t message_len) {
   std::string message{message_ptr, (size_t)message_len};
   std::cout << message << std::flush;
 }
+
+void *concrete_checked_malloc(size_t size) {
+  void *ptr = malloc(size);
+  if (ptr == nullptr) {
+    // print error and backtrace to stderr
+    fprintf(stderr, "bad alloc: nullptr while calling malloc(%zu)\n", size);
```

Comment:
maybe for human readiness, I would also write the size in, let say GB
it's much easier to understand 

   "bad alloc: nullptr while calling malloc(18446744073 GB)"

than 

   "bad alloc: nullptr while calling malloc(18446744073709551615)"

Reply:
what if a bad malloc happens with a size < 1GB for some reason? We can write different things based on the value, but I would think this is easiest for debugging because (1) if you don't want to use a debugger, you can just see which one is doing the allocation with that exact value (2) doing conversion is easy

Reply:
then you would write a function which write:
- the number of KB if less than 1 MB
- the number of MB if less than 1 GB
- or the number of GB
something like that. 

Doing the conversion is easy yes, but often users don't make any effort. If we tell them they ask for 10000000 GB, I hope it's clear to them. Reporting to users in an easy way is important, I think, it saves our time as well, it avoids reports to which we can't do anything.